### ISSUE TITLE
Is it possible to update the code with kubernetes 1.18?

### ISSUE THREAD

**Issue by cf250024** [programming-kubernetes/custom-apiserver#4]
I guess the code has been tested with api-server nightly build. Since then, there are changes in api-server. I see some errors when I tried to use 1.21 and then 1.18 as well? I am trying to make change locally while reading through the book. But I doubt it could be running without issues after I make changes.

Is it possible to update the code so that it could be compiled and run with api-server 1.18? This will make sure the issues are on us if it can't be running with our local tests. Thanks.

**Comment by mhausenblas** [comment]
SGTM, wanna PR it?

**Comment by cf250024** [comment]
Yes, please. (is not this PR https://github.com/programming-kubernetes/pizza-apiserver/pull/3, correct?) Thanks.
